configuration-service/common/git_test.go

```diff
@@ -786,3 +795,55 @@ func TestGit_CloneRepo(t *testing.T) {
 		})
 	}
 }
+
+func Test_unmarshalGitCredentials(t *testing.T) {
+	type args struct {
+		data []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *common_models.GitCredentials
+		wantErr bool
+	}{
+		{
+			name: "get secret",
+			args: args{
+				data: []byte(`{"user":"user","token":"token","remoteURI":"http://remote-url.git"}`),
+			},
+			want: &common_models.GitCredentials{
+				User:      "user",
+				Token:     "token",
+				RemoteURI: "http://remote-url.git",
+			},
+			wantErr: false,
+		},
+		{
+			name: "no token and remote URI",
+			args: args{
+				data: []byte(`{"user":"user","token":"","remoteURI":""}`),
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "malformed credentials",
+			args: args{
+				data: []byte(`invalid`),
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
please add test for input: `{"token":"token","remoteURI":"http://remote-url.git"}`

Reply:
done